security/security-spring/src/main/resources/applicationContext-security.xml

```diff
@@ -401,7 +401,19 @@
     </b:bean>
 
     <!-- Provider of default SAML Context -->
-    <b:bean id="contextProvider" class="org.springframework.security.saml.context.SAMLContextProviderImpl"/>
+    <b:bean id="contextProvider" class="org.springframework.security.saml.context.SAMLContextProviderImpl">
+        <b:property name="storageFactory">
+            <b:bean class="${saml.context:org.springframework.security.saml.storage.HttpSessionStorageFactory}"/>
+        </b:property>
+    </b:bean>
+
+    <!--
+        RelayState success handler to process return URL for IDP initiated login SAML response.
+        Application landing page is going to be RelayState URL upon successfull authentication.
+    -->
+    <b:bean id="successRedirectHandler" class="org.springframework.security.saml.SAMLRelayStateSuccessHandler">
+        <b:property name="defaultTargetUrl" value="/" />
```

Comment:
Should this defaultTargetUrl be consistent with [successRedirectHandler](https://github.com/cBioPortal/cbioportal/blob/6a5fbbbb141a4850fd36f10222ff81592c5e5484/security/security-spring/src/main/resources/applicationContext-security.xml#L125).defaultTargetUrl ?

Reply:
Yes, it is. Application behavior remains the same for defaultTargetURL. The application always redirects to the user requested URL upon successful login. SAMLRelayStateSuccessHandler additionally process relayState property URL part of SAML response from the IDP.